src/Exchange.ts

```diff
@@ -0,0 +1,435 @@
+          takerFee: utils.formatAmount(b.takerFee, baseDecimals)
+        }))
+      }
+    } catch (error) {
+      return Promise.reject(error)
+    }
+  }
+
+  public async makeOrder (
+    exchangeContractAddress: string,
+    makerTokenAddress: string,
+    takerTokenAddress: string,
+    makerTokenValue: number | string,
+    takerTokenValue: number | string,
+    {
+      makerTokenDecimals,
+      takerTokenDecimals,
+      expirationUnixTimestampSec = 2524604400
+    }: ExchangeOptions = {}
+  ): Promise<any> {
+    const { currencyNetwork, user, utils } = this
+    try {
+      const [ makerDecimals, takerDecimals ] = await Promise.all([
+        currencyNetwork.getDecimals(makerTokenAddress, makerTokenDecimals),
+        currencyNetwork.getDecimals(takerTokenAddress, takerTokenDecimals)
+      ])
+      const feesRequest = {
+        exchangeContractAddress,
+        expirationUnixTimestampSec: expirationUnixTimestampSec.toString(),
+        maker: user.address,
+        makerTokenAddress: ethUtils.toChecksumAddress(makerTokenAddress),
+        makerTokenAmount: utils.calcRaw(makerTokenValue, makerDecimals).toString(),
+        salt: Math.floor(Math.random() * 1000000000).toString(),
+        taker: ZERO_ADDRESS,
+        takerTokenAddress: ethUtils.toChecksumAddress(takerTokenAddress),
+        takerTokenAmount: utils.calcRaw(takerTokenValue, takerDecimals).toString()
+      }
+      const { feeRecipient, makerFee, takerFee } = await this.getFees(feesRequest)
+      const order = {
+        ...feesRequest,
+        makerFee,
+        takerFee,
+        feeRecipient
+      }
+      const orderHash = this.getOrderHashHex(order)
+      return user.signMsg(orderHash)
+        .then(({ ecSignature }) => ({...order, ecSignature}))
+        .then(signedOrder => this.postRequest('exchange/order', signedOrder)
+          .then(() => ({
+            ...signedOrder,
+            hash: orderHash,
+            makerTokenAmount: utils.formatAmount(signedOrder.makerTokenAmount, makerDecimals),
+            takerTokenAmount: utils.formatAmount(signedOrder.takerTokenAmount, takerDecimals)
+          }))
+      )
+    } catch (error) {
+      return Promise.reject(error)
+    }
+  }
+
+  /**
+   * Prepares an on-chain transaction for a fill order.
+   * @param orderAddresses Array of order's maker, taker, makerToken, takerToken, and feeRecipient.
+   * @param orderValues Array of order's makerTokenAmount, takerTokenAmount, makerFee, takerFee, expirationTimestampInSec, and salt.
+   * @param fillTakerTokenAmount Desired amount of takerToken to fill.
+   * @param makerPath Path in the currency network of the maker token or [] if no currency network
+   * @param takerPath Path in the currency network of the taker token or [] if no currency network
+   * @param v ECDSA signature parameter v.
+   * @param r ECDSA signature parameter r.
+   * @param s ECDSA signature parameter s.
+   */
+  public async prepTakeOrder (
```

Comment:
Can we change the the signature of the function to

prepTakeOrder(signedOrder: singedOrder)

Reply:
yes wanted to do this anyways